### PR TITLE
Pam 21

### PR DESCRIPTION
**Pull request by saugat977**
PAM-21: Implemented logout user functionality
Workflow:
- User clicks on logout button
- The system retrieves token from localstorage
- API is called for logout
- the token in localstorage is cleared
-  Redirected to login page after successful logout

### CODE REVIEW

File: src/components/DashBoard.vue [hunk 1]
```diff
@@ -216,50 +252,48 @@ body.sb-sidenav-toggled #wrapper #sidebar-wrapper {
...
-  
 }
 
-.dashboard-menu{
-  border:none !important;
-  padding:15px !important;
-  text-decoration:none;
-  color:black;
+.dashboard-menu {
+  border: none !important;
```
Comment (asimnp) [inline src/components/DashBoard.vue]: Why is it required to add "!important"? Can't we remove it?

File: src/components/DashBoard.vue [hunk 2]
```diff
@@ -216,50 +252,48 @@ body.sb-sidenav-toggled #wrapper #sidebar-wrapper {
...
+  height: 35px;
+  width: 35px;
+  border: 1px solid grey;
   border-radius: 50%;
-
 }
-.nav-link{
-  position:relative !important;
+.nav-link {
+  position: relative !important;
```
Comment (asimnp) [inline src/components/DashBoard.vue]: Why is it required to add "!important"? Can't we remove it?

File: src/components/DashBoard.vue [hunk 3]
```diff
@@ -216,50 +252,48 @@ body.sb-sidenav-toggled #wrapper #sidebar-wrapper {
...
 }
 /* .dashboard-menu:hover {
  color: #FA6731 ; 
 } */
-.router-link-exact-active{
-  border-radius:5px !important;
-  color:white !important;
-  background-color: #FA6731 ;
+.router-link-exact-active {
+  border-radius: 5px !important;
```
Comment (asimnp) [inline src/components/DashBoard.vue]: Why is it required to add "!important"? Can't we remove it?

File: src/modules/account/utils/logout.js [hunk 4]
```diff
@@ -1,5 +1,17 @@
...
-    //should expire cookie 
-    // remove the login data from local storage[dependent on login]
-localStorage.removeItem('user')
+import {logoutUser} from '../services'
+import getToken from '@/utils/getToken';
+export default async function logout(){
+    const token = getToken()
+    if (!token){
+        //show error message in a dialog box or a toast
+        console.log('no token found');
```
Comment (asimnp) [inline src/modules/account/utils/logout.js]: Remove console.log

File: src/modules/account/utils/logout.js [hunk 5]
```diff
@@ -1,5 +1,17 @@
...
+        return false
+
+    }
+    const response = await logoutUser(token)
+    if (response.data.status.toLowerCase() != 'success'){
+    //use toast to show error
```
Comment (asimnp) [inline src/modules/account/utils/logout.js:12]: Why there is nothing inside the if block?

Comment (saugat977) [reply src/modules/account/utils/logout.js:12]: We're planning on implementing toast messages for every success and error message on the project, hence the task inside the if block will fall under a different card.

File: src/components/DashBoard.vue (hunk 3 above)
Comment (saugat977) [reply src/components/DashBoard.vue]: removed

File: src/components/DashBoard.vue (hunk 2 above)
Comment (saugat977) [reply src/components/DashBoard.vue]: removed

File: src/components/DashBoard.vue (hunk 1 above)
Comment (saugat977) [reply src/components/DashBoard.vue]: removed

File: src/modules/account/utils/logout.js (hunk 4 above)
Comment (saugat977) [reply src/modules/account/utils/logout.js]: removed

File: src/components/DashBoard.vue [hunk 6]
```diff
@@ -216,52 +231,51 @@ body.sb-sidenav-toggled #wrapper #sidebar-wrapper {
...
+  padding: 15px ;
+  text-decoration: none;
+  color: black;
+}
+.dashboard-menu:hover {
+ color: #FA6731 ; 
+}
+.router-link-exact-active {
+  border-radius: 5px ;
+  color: white !important;
```
Comment (asimnp) [inline src/components/DashBoard.vue]: Why there is "!important"?

**Pull request merged**